File: server_asp/src/models/Message.cs
using System;
using System.Collections.Generic;

namespace shieldtify.models;

public partial class Message
{
    public Guid Uid { get; set; }

    public Guid Conversationid { get; set; }

    public string Content { get; set; } = null!;

    public string Role { get; set; } = null!;

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public DateTime? DeletedAt { get; set; }

    public virtual Conversation Conversation { get; set; } = null!;
}

File: server_nodejs/src/models/message.js
// Message (UID, conversationID, content, role)

module.exports = (sequelize, Sequelize) => {
    class Message extends Sequelize.Model {}

    Message.init(
        {
            uid: {
                type: Sequelize.UUID,
                primaryKey: true,
            },
            conversationid: {
                type: Sequelize.UUID,
                allowNull: false,
            },
            content: {
                type: Sequelize.TEXT('long'),
                allowNull: false,
            },
            role: {
                type: Sequelize.STRING,
                allowNull: false,
            },
            index: {
                type: Sequelize.INTEGER,
                allowNull: false,
            },
        },
        {
            sequelize,
            modelName: 'message',
            timestamps: true,
            paranoid: false,
            underscored: true,
        },
    );

    Message.associate = (models) => {
        Message.belongsTo(models.conversation, {
            foreignKey: 'conversationid',
        });
    };
    return Message;
};
